usr/share/lib/ipa/tests/conftest.py

```diff
@@ -39,6 +34,43 @@ def f(repo):
     return f
 
 
+@pytest.fixture()
+def determine_provider(host):
+    def f():
+        result = host.run('sudo dmidecode -t system')
+        output = result.stdout.lower()
+        if 'amazon' in output:
+            provider = 'ec2'
+        elif 'microsoft' in output:
+            provider = 'azure'
+        elif 'google' in output:
+            provider = 'gce'
+        else:
+            raise Exception('Provider not found.')
+        return provider
+    return f
+
+
+@pytest.fixture()
+def determine_region(host):
+    def f(provider):
+        if provider == 'ec2':
+            result = host.run('ec2metadata --availability-zone')
+            region = result.stdout.strip()[:-1]
+        elif provider == 'gce':
+            result = host.run('gcemetadata --query instance --zone')
+            region = result.stdout.strip().rsplit('/', maxsplit=1)[-1]
```

Comment:
The implementation for ec2 and gce forces the user of IPA to install these packages. Which is fine with me. We'll see if other projects pick up IPA and if so I would be OK converting this to direct API calls.

Leave as is for now, just something to keep in mind.

Reply:
Okay, will note this.